modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/fork/ForkInfoStorageSuite.scala

```diff
@@ -0,0 +1,90 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import cats.effect.{IO, Ref}
+
+import scala.collection.immutable.Queue
+
+import org.tessellation.schema.generators.{peerIdGen, snapshotOrdinalGen}
+import org.tessellation.sdk.config.types.ForkInfoStorageConfig
+import org.tessellation.sdk.domain.fork.{ForkInfo, ForkInfoStorage, StoredForkInfo}
+import org.tessellation.security.hash.Hash
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.cats.refTypeShow
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.types.numeric.PosInt
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object ForkInfoStorageSuite extends SimpleIOSuite with Checkers {
+  val genStoredForkInfoEntry: Gen[StoredForkInfo] = for {
+    peerId <- peerIdGen
+    ordinal <- snapshotOrdinalGen
+    hashString <- Gen.alphaStr
+    hash = Hash(hashString)
+  } yield StoredForkInfo(peerId, ForkInfo(ordinal, hash))
+
+  def mkForkInfoStorage(queue: Queue[StoredForkInfo], queueSize: PosInt): F[ForkInfoStorage[ForkInfoStorageSuite.F]] = {
+    val config = ForkInfoStorageConfig(queueSize)
+
+    Ref[IO].of(queue).map(ForkInfoStorage.make(_, config))
+  }
+
+  test("initial store is empty") {
+    val gen = for {
+      queueSize <- Gen.chooseNum(2, 10)
+      refinedQueueSize = Refined.unsafeApply[Int, Positive](queueSize)
+    } yield refinedQueueSize
+
+    forall(gen) { queueSize =>
+      for {
+        store <- mkForkInfoStorage(Queue.empty, queueSize)
+
+        expected = List.empty[StoredForkInfo]
+        actual <- store.getForkInfo
+      } yield expect.eql(expected, actual.toList)
+    }
+  }
+
+  test("successfully add when there is capacity") {
+    val gen = for {
+      entry <- genStoredForkInfoEntry
+      queueSize <- Gen.chooseNum(2, 10)
+      refinedQueueSize = Refined.unsafeApply[Int, Positive](queueSize)
+    } yield (entry, refinedQueueSize)
+
+    forall(gen) {
+      case (entry, queueSize) =>
+        for {
+          store <- mkForkInfoStorage(Queue.empty, queueSize)
+
+          _ <- store.add(entry)
+
+          expected: List[StoredForkInfo] = List(entry)
+          actual <- store.getForkInfo.map(_.toList)
+        } yield expect.eql(expected, actual)
+    }
+  }
+
+  test("add when at capacity ejects the first queue entry") {
+    val gen = for {
+      size <- Gen.chooseNum(1, 10)
+      entries <- Gen.listOfN(size, genStoredForkInfoEntry)
+      queueSize = Refined.unsafeApply[Int, Positive](size)
+      newEntry <- genStoredForkInfoEntry
+    } yield (entries, queueSize, newEntry)
+
+    forall(gen) {
+      case (entries, queueSize, entry) =>
+        for {
+          store <- mkForkInfoStorage(Queue.from(entries), queueSize)
+
+          _ <- store.add(entry)
+
+          expected: List[StoredForkInfo] = entries.tail :+ entry
+          actual <- store.getForkInfo.map(_.toList)
+        } yield expect.eql(expected, actual)
+    }
+  }
```

Comment:
Food for thought: Do we need two tests for this?

Couldn't we do this with a single test where the number of entries generated is anywhere from 0 to _n * queueSize_ where _n = 2 or 3_ ?  (we really don't need anything big)

(Incidentally, shouldn't that variable be called `storageSize`?)

To verify the storage is working as expected you'd check that the sequence returned from the storage contains the last `storageSize` values input, which covers but under and overflow.

Reply:
Consolidated the two tests.